### PR TITLE
Patch for v1.0.6

### PR DESCRIPTION
**Pull request by JoeWaddell**
### Context

This PR contains 3 patches:

1. Prevents error generated in farmgrown feed emissions methods for the edge case when simulating fields without crops being grown.
2. Prevents a divide by zero error in soil water content methods that is already fixed on `dev`.
3. Prevents repro Enum state error during initialization of cows entering the herd in a pregnant state. See PR #3263
4. Removes the deepdiff comparison of results from e2e test simulation runs from old expected results in the step prior to overwriting those results as the new expected e2e results. Implementing on `dev` branch via #3275.
5. Fixes an issue where the tolerance set for E2E testing results comparison was not being properly implemented. Implemented on `dev` branch in #3026.

### Test plan

- Test on edge case farms that produced the error
- Test on subset of evaluation farms to ensure behavior did not change.
- Test that e2e expected results can be updated without exorbitant runtime.
- Test that e2e testing is passing.

### Input Changes

None.

### Output Changes

None.

#### Filter

```json

```

### CODE REVIEW

File: RUFAS/biophysical/field/crop/crop_management.py [hunk 1]
```diff
@@ -487,7 +487,7 @@ def _distribute_residue_nutrients(self, soil_data: SoilData) -> None:
 
         """
         surface_layer = soil_data.soil_layers[0]
-        surface_fraction = (self.yield_residue - self.data.root_biomass) / self.yield_residue
+        surface_fraction = ((self.yield_residue - self.data.root_biomass) / self.yield_residue) if self.yield_residue > 0.0 else 0.0
```
Comment (PooyaHekmati) [inline RUFAS/biophysical/field/crop/crop_management.py]: better readability
```suggestion
        
        if self.yield_residue > 0.0
              surface_fraction = (self.yield_residue - self.data.root_biomass ) / self.yield_residue
         else:
                surface_fraction = 0
```

**Review by morrowcj — APPROVED** [review]
LGTM, but we should make sure to add unit tests when we add this bug fix to dev.

**Comment by KFosterReed** [comment]
I tested this on farms 2, 5, 6, and 7 and, as expected, results did not change beyond those due to random variation.

**Review by KFosterReed — APPROVED** [review]
Works as expected!

**Review by allisterakun — CHANGES_REQUESTED** [review]
Changes look good, but some e2e tests are failing:

> Summary of e2e results:
        
    freestall_e2e results:
      Animal: Failing
      CropAndSoil: Passing
      Manure: Passing
      Feed: Passing
    
    open_lot_e2e results:
      Animal: Failing
      CropAndSoil: Failing
      Manure: Failing
      Feed: Failing
    
    no_animal_e2e results:
      Animal: n/a
      CropAndSoil: Passing
      Manure: Passing
      Feed: Passing


I have tried on both my Mac and Linux machine, all failing at the same modules

**Review by allisterakun — APPROVED** [review]
